tests/integration/tests/test_etcd.py

```diff
@@ -42,3 +44,60 @@ def test_etcd(instances: List[harness.Instance], etcd_cluster: EtcdCluster):
         ["systemctl", "is-active", "--quiet", "snap.k8s.k8s-dqlite"], check=False
     )
     assert p.returncode != 0, "k8s-dqlite service is still active"
+
+    LOG.info("Add new etcd nodes")
+    etcd_cluster.add_nodes(2)
+
+    # Update  server-urls in cluster
+    body = {
+        "datastore": {
+            "type": "external",
+            "servers": etcd_cluster.client_urls,
+            "ca-crt": etcd_cluster.ca_cert,
+            "client-crt": etcd_cluster.cert,
+            "client-key": etcd_cluster.key,
+        }
+    }
+    k8s_instance.exec(
+        [
+            "curl",
+            "-XPUT",
+            "--header",
+            "Content-Type: application/json",
+            "--data",
+            json.dumps(body),
+            "--unix-socket",
+            "/var/snap/k8s/common/var/lib/k8sd/state/control.socket",
+            "http://localhost/1.0/k8sd/cluster/config",
+        ]
+    )
+
+    # Give some time to reconcile and restart services.
+    time.sleep(2)
+
+    # check that we can still connect to the kubernetes cluster
+    k8s_instance.exec(["k8s", "kubectl", "get", "pods", "-A"])
+
+    # Changing the datastore back to k8s-dqlite after using the external datastore should fail.
+    body = {
+        "datastore": {
+            "type": "k8s-dqlite",
+        }
+    }
+
+    resp = k8s_instance.exec(
+        [
+            "curl",
+            "-XPUT",
+            "--header",
+            "Content-Type: application/json",
+            "--data",
+            json.dumps(body),
+            "--unix-socket",
+            "/var/snap/k8s/common/var/lib/k8sd/state/control.socket",
+            "http://localhost/1.0/k8sd/cluster/config",
+        ],
+        capture_output=True,
+    )
+    response = json.loads(resp.stdout.decode())
+    assert response["error_code"] == 400, "changing the datastore type should fail"
```

Comment:
i would expect this should be a unit test instead

Reply:
true, I have added a unittest for the converter.
But I also keep this check. It is quick - the cluster is up already -  and we don't have a good way to test the API.

Reply:
sounds good